cs3api4lab/api/cs3_ocm_share_api.py

```diff
@@ -0,0 +1,268 @@
+            self._handle_error(response)
+        return self._map_ocm_shares(response)
+
+    def _get_ocm_share(self, share_id):
+        share_id_obj = sharing_res.ShareId(opaque_id=share_id)
+        ref = sharing_res.ShareReference(id=share_id_obj)
+        request = ocm_api.GetOCMShareRequest(ref=ref)
+        response = self.ocm_share_api.GetOCMShare(request=request, metadata=self._token())
+        return self._map_ocm_share(response.share)
+
+    def get_received_ocm_shares(self, share_id):
+        if share_id is None:
+            return self._list_received_shares()
+        else:
+            return self._get_received_share(share_id)
+
+    def _list_received_shares(self):
+        request = ocm_api.ListReceivedOCMSharesRequest()
+        response = self.ocm_share_api.ListReceivedOCMShares(request=request,
+                                                            metadata=self._token())
+        if not self._is_code_ok(response):
+            self.log.error("Error listing OCM received shares:")
+            self._handle_error(response)
+        return self._map_ocm_shares(response, received=True)
+
+    def _get_received_share(self, share_id):
+        share_id_obj = sharing_res.ShareId(opaque_id=share_id)
+        ref = sharing_res.ShareReference(id=share_id_obj)
+        request = ocm_api.GetReceivedOCMShareRequest(ref=ref)
+        response = self.ocm_share_api.GetReceivedOCMShare(request=request,
+                                                          metadata=self._token())
+        if not self._is_code_ok(response):
+            self.log.error("Error listing OCM received share:")
+            self._handle_error(response)
+        return self._map_ocm_received_share(response.share)
+
+    def _map_ocm_shares(self, list_response, received=False):
+        shares = []
+        if received:
+            for share in list_response.shares:
+                shares.append(self._map_ocm_received_share(share))
+        else:
+            for share in list_response.shares:
+                shares.append(self._map_ocm_share(share))
+        return shares
+
+    def _map_ocm_share(self, share):
+        return {
+            "id": share.id.opaque_id,
+            "storage_id": share.resource_id.storage_id,
+            "opaque_id": share.resource_id.opaque_id,
+            "name": share.name,
+            "permissions": ShareUtils.map_permissions_to_role(share.permissions.permissions),
+            "grantee": {
+                "type": ShareUtils.map_grantee_type(share),
+                "idp": share.grantee.id.idp,
+                "opaque_id": share.grantee.id.opaque_id
+            },
+            "owner": {
+                "idp": share.owner.idp,
+                "opaque_id": share.owner.opaque_id
+            },
+            "creator": {
+                "idp": share.creator.idp,
+                "opaque_id": share.creator.opaque_id
+            }
+        }
+    
+    def _map_ocm_received_share(self, share):
```

Comment:
Is it me, or this is exactly the same as `_map_ocm_share` with an extra `state`? I think we could avoid repetition..

Reply:
I fixed it